easybuild/easyblocks/c/comsol.py

```diff
@@ -0,0 +1,133 @@
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for installing COMSOL, implemented as an easyblock
+
+@author: Mikael OEhman (Chalmers University of Technology)
+@author: Ake Sandgren (HPC2N, Umea University)
+"""
+import re
+import os
+import shutil
+import stat
+
+import easybuild.tools.environment as env
+from easybuild.easyblocks.generic.packedbinary import PackedBinary
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, copy_file, find_flexlm_license, read_file, write_file
+from easybuild.tools.modules import get_software_root
+from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_COMSOL(PackedBinary):
+    """Support for installing COMSOL."""
+
+    def __init__(self, *args, **kwargs):
+        """Add extra config options specific to COMSOL."""
+        super(EB_COMSOL, self).__init__(*args, **kwargs)
+        self.comp_fam = None
+        self.configfile = os.path.join(self.builddir, 'my_setupconfig.ini')
+
+    def configure_step(self):
+        """Configure COMSOL installation: create license file."""
+
+        # The tar file comes from the DVD and has 0444 as permission at
+        # the top dir.
+        adjust_permissions(self.start_dir, stat.S_IWUSR)
+
+        default_lic_env_var = 'LMCOMSOL_LICENSE_FILE'
+        lic_specs, self.license_env_var = find_flexlm_license(custom_env_vars=[default_lic_env_var],
+                                                              lic_specs=[self.cfg['license_file']])
+
+        if lic_specs:
+            if self.license_env_var is None:
+                self.log.info("Using COMSOL license specifications from 'license_file': %s", lic_specs)
+                self.license_env_var = default_lic_env_var
+            else:
+                self.log.info("Using COMSOL license specifications from $%s: %s", self.license_env_var, lic_specs)
+
+            self.license_file = os.pathsep.join(lic_specs)
```

Comment:
should be `os.path.join`?

Reply:
yes clearly. probably C&P bug

Reply:
Or maybe not? intelbase.py also uses pathsep.join for the same purpose. And so are all the other blocks that uses find_flexlm_license. So I'll leave it as is to avoid breaking anything. path.join and join doesn't work exactly the same.

Reply:
Ah, I see why, with `os.path.join` you would have to 'unpack' `lic_specs` since `os.path.join` requires multiple arguments rather than one single list argument, which is uglier:

```python
os.path.join(*lic_specs)
```

So do leave as is, indeed.